src/proficiency.cpp

```diff
@@ -220,6 +221,22 @@ std::vector<proficiency_bonus> proficiency::get_bonuses( const std::string &cate
     return bonus_it->second;
 }
 
+std::optional<float> proficiency::bonus_for( const std::string &category,
```

Comment:
Why `std::optional<float>`?

You could just return `float`.
Instead of `nullptr` return 0.

Reply:
This is generic infrastructure. While that would be an appropriate value for my use case, if the bonus value is legitimately zero, that makes this function useless.

Reply:
> This is generic infrastructure.
Where else will it be used? When not returning 0 makes sense?

Reply:
I don't know where else it will be used, and that's why I elected to make it optional instead of returning zero.

Reply:
If you cannot think of anything else it is not likely to ever need to be nullptr. It is going to be either a positive float, then add it, or not affect, then make it zero, which is neutral to addition. If somebody really needs to make it different, they can simply refactor it.

I care because it is not free abstraction. It makes the code more complicated for no reason.